modules/cli/src/main/scala/scala/cli/commands/test/TestOptions.scala

```diff
@@ -6,14 +6,27 @@ import caseapp.core.help.Help
 import scala.cli.commands.shared.{
   CrossOptions,
   HasSharedOptions,
+  HelpMessages,
   SharedJavaOptions,
   SharedOptions,
   SharedWatchOptions
 }
 import scala.cli.commands.tags
 
 // format: off
-@HelpMessage("Compile and test Scala code")
+@HelpMessage(s"""Compile and test Scala code.
+                |
+                |Test sources are compiled separately (after the 'main' sources), and may use different dependencies, compiler options, and other configurations.
+                |A source file is treated as a test source if:
+                |  - it contains the `//> using target.scope "test"` directive
+                |  - the file name ends with `.test.scala`
+                |  - the file comes from a directory that is provided as input, and the relative path from that file to its original directory contains a `test` directory
```

Comment:
```suggestion
                |  - the file comes from a directory that is provided as input, and the relative path from that file to its workspace directory contains a `test` directory
```

Reply:
I'd be careful about calling it the `workspace`directory, as we call the `.scala-build` directory that elsewhere: https://scala-cli.virtuslab.org/docs/reference/cli-options/#workspace-options